pymbolic/mapper/c_code.py

```diff
@@ -72,33 +84,47 @@ class CCodeMapper(SimplifyingSortingStringifyMapper):
     for the ``cse_*`` attributes.
     """
 
-    def __init__(self, reverse=True,
-            cse_prefix="_cse", complex_constant_base_type="double",
-            cse_name_list=None):
+    cse_prefix: str
+    cse_to_name: dict[Expression, str]
+    cse_names: set[str]
+    cse_name_list: list[tuple[str, str]]
+    complex_constant_base_type: str
```

Comment:
These type annotations are not quite right, but I'm not sure how this is supposed to work. The main issue is that
```
cse_to_name = {cse: name for name, cse in cse_name_list}
```
which would mean that `cse_name_list: Sequence[tuple[str, Expression]]`, but in `map_common_subexpression`, we just add strings to both this and `cse_names`. So something's not quite right in here.. I couldn't find a usage that actually passes in `cse_name_list` either.